File: src/programmingwithdocuments/workingwithfields/removefield/java/RemoveField.java
/*
 * Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
 *
 * This file is part of Aspose.Words. The source code in this file
 * is only intended as a supplement to the documentation, and is provided
 * "as is", without warranty of any kind, either expressed or implied.
 */

package programmingwithdocuments.workingwithfields.removefield.java;

import com.aspose.words.*;

public class RemoveField
{
    public static void main(String[] args) throws Exception
    {
        // The path to the documents directory.
        String dataDir = "src/programmingwithdocuments/workingwithfields/removefield/data/";

        Document doc = new Document(dataDir + "RemoveField.doc");

        //ExStart
        //ExFor:Field.Remove
        //ExId:DocumentBuilder_RemoveField
        //ExSummary:Removes a field from the document.
        Field field = doc.getRange().getFields().get(0);
        // Calling this method completely removes the field from the document.
        field.remove();
        //ExEnd
    }
}

File: Plugins/Aspose_Words_Java_for_Ruby/lib/asposewordsjavaforruby/removefield.rb
module Asposewordsjavaforruby
  module RemoveField
    def initialize()
        # The path to the documents directory.
        data_dir = File.dirname(File.dirname(File.dirname(__FILE__))) + '/data/'

        # Open the document.
        doc = Rjb::import('com.aspose.words.Document').new(data_dir + "Field.RemoveField.doc")

        #ExStart
        #ExFor:Field.Remove
        #ExId:DocumentBuilder_RemoveField
        #ExSummary:Removes a field from the document.
        field = doc.getRange().getFields().get(0)
        # Calling this method completely removes the field from the document.
        field.remove()
        #ExEnd

        # Save the document.
        doc.save(data_dir + "Field.RemoveField Out.doc")
    end
  end
end
